### PR TITLE
Add build file tree functionality.

### PR DESCRIPTION
**Pull request by hacktobeer**
Adds functionality to build a file tree on disk from the original_path attribute from an array of OutputFile objects. This way workers can mimic OutputFiles in their original folder. This can be useful for several workers that want/need to work on original folder/filename on disk tree structures.

### CODE REVIEW

File: openrelik_worker_common/utils.py [hunk 1]
```diff
@@ -187,3 +189,67 @@ def create_output_file(
...
+    """Creates the original file tree structure from a list of OutputFiles.
+
+    Args:
+        files: A list of OutPutFile instances.
+
+    Returns:
+        The root path of the file tree as a TemporaryDirectory.
+    """
+    os_tmpdir = tempfile.gettempdir()
+    tree_root = tempfile.TemporaryDirectory(delete=False)
```
Comment (berggren) [inline openrelik_worker_common/utils.py:217]: Using tempfile here won't work when the artifacts are stored on another disk than the OS. For example we will not be able to hard link files between and NFS mount (e.g. filestore) and OS:/tmp/ 

In the workers we create a temporary directory manually on the same path as where the artifacts are stored:

Example:
```
# Create temporary directory and hard link files for processing
temp_dir = os.path.join(output_path, uuid4().hex)
os.mkdir(temp_dir)
for input_file in input_files:
    filename = os.path.basename(input_file.get("path"))
    os.link(input_file.get("path"), f"{temp_dir}/{filename}")

... do you stuff ...

# Remove the temp_dir
shutil.rmtree(temp_dir)
```

**Review by berggren — CHANGES_REQUESTED** [review]
Good stuff. Some comments around the use of tempfile and hardlinks

File: openrelik_worker_common/utils.py [hunk 2]
```diff
@@ -187,3 +189,67 @@ def create_output_file(
...
+    """Converts a full path to relative path without the root.
+
+    Args:
+        path: A full path.
+
+    Returns:
+        A relative path without the root.
+    """
+    # TODO(hacktobeer) Make this cross-platform
+    return str(PurePath(path).relative_to('/'))
```
[the remark this answers is not in the document]
Comment (hacktobeer) [reply openrelik_worker_common/utils.py]: nice

File: openrelik_worker_common/utils.py [hunk 3]
```diff
@@ -187,3 +189,67 @@ def create_output_file(
...
+    for file in files:
+        normalized_path = os.path.normpath(file.original_path)
+        original_filename = Path(normalized_path).name
+        original_folder = Path(normalized_path).parent
+        relative_original_folder = get_path_without_root(original_folder)
+        #Create full folder
+        try:
+            tmp_full_path = os.path.join(tree_root.name,
+                                         relative_original_folder)
+            if os_tmpdir not in tmp_full_path:
```
[the remark this answers is not in the document]
Comment (hacktobeer) [reply openrelik_worker_common/utils.py]: ack

File: openrelik_worker_common/utils.py (hunk 1 above)
Comment (hacktobeer) [reply openrelik_worker_common/utils.py]: Makes sense. I still like a TemporaryDirectory instance to use in the delete_file_tree function but have created it with the dir parameter to make sure it is create in the OpenRelik configured output_path location.

I have also plumbed through the output_path in a non OutputFile generic way. As the run function in the worker gets this as a configured parameter.

**Comment by hacktobeer** [comment]
Thanks for the comments, PTAL.

File: openrelik_worker_common/utils.py [hunk 4]
```diff
@@ -187,3 +189,77 @@ def create_output_file(
...
+        return None
+
+    tree_root = tempfile.TemporaryDirectory(dir=output_path, delete=False)
+
+    for file in files:
+        normalized_path = os.path.normpath(file.original_path)
+        original_filename = Path(normalized_path).name
+        original_folder = Path(normalized_path).parent
+        relative_original_folder = get_path_without_root(original_folder)
+        #Create full folder
```
Comment (berggren) [inline openrelik_worker_common/utils.py:229]:
```suggestion
        # Create complete folder structure.
```

**Review by berggren — APPROVED** [review]
LGTM

**Pull request merged**